common/membership/ringpop/monitor.go

```diff
@@ -153,13 +172,50 @@ func (rpo *monitor) Start() {
 		rpo.logger.Fatal("unable to set ring pop ServiceRole label", tag.Error(err))
 	}
 
+	// Our individual rings may not support concurrent start/stop calls so we reacquire the state lock while acting upon them
+	rpo.stateLock.Lock()
 	for _, ring := range rpo.rings {
 		ring.Start()
 	}
+	rpo.stateLock.Unlock()
 
 	rpo.initialized.Set(struct{}{}, nil)
 }
 
+// bootstrap ring pop service by discovering the bootstrap hosts and joining the ring pop cluster
+func (rpo *monitor) bootstrapRingPop(
+	bootstrapHostPostRetriever func() ([]string, error),
```

Comment:
this can be simplified more.. just call `rpo.fetchCurrentBootstrapHostports` directly and use `healthyHostLastHeartbeatCutoff/2` for the backoff interval